ttm_module_publish.gradle

```diff
@@ -0,0 +1,66 @@
+enum BuildType {
+    DEVELOP("-SNAPSHOT"), BETA("-BETA"), RELEASE(""), FEATURE("-FEATURE")
+
+    BuildType(String classifier) { this.classifier = classifier }
+    private final String classifier
+
+    String toString() { return classifier }
+
+    static fromBranch(String branch) {
+        // We use git flow - http://nvie.com/posts/a-successful-git-branching-model/
+        // The defaults use release/, hotfix/, etc instead of release- and hotfix- mentioned above
+        if (branch.equals("master")) {
+            //Our only releases are on master
+            return RELEASE
+        } else if (branch.startsWith("release/") || branch.startsWith("hotfix/")) {
+            //Both releases and hotfixes are prereleases - a new version that will show up as a full release later
+            return BETA
+        } else if (branch.equals("develop")) {
+            return DEVELOP
+        } else {
+            return FEATURE
+        }
+    }
+}
+
+apply plugin: 'maven-publish'
+
+ext {
+    buildType = BuildType.fromBranch(project.findProperty('branch') ?: 'feature')
+    artifactoryUrl = 'https://artifactory.trimble.tools/artifactory'
+    publishRepositoryKey = "ttm-mobile-ecosystem-${buildType == BuildType.RELEASE ? 'release' : 'snapshot'}-maven"
+}
+
+publishing {
+    publications {
+        aar(MavenPublication) {
+            groupId 'com.trimble'
+            version = "${rootProject.version}${buildType.toString()}"
+            artifactId "${rootProject.name}-${project.getName()}"
```

Comment:
Does `rootProject.name` work on the build server? [It seems that `rootProject.name` is based on the name of the directory that the build.gradle file is in](https://mrhaki.blogspot.com/2009/11/gradle-goodness-changing-project-name.html), but that's not actually committed to the repository, so we can't control what that is. We should either explicitly define the name or we should have the directory as part of the repository. Both have downsides. What are your thoughts?

Reply:
My bad there. How about we use the repo name from bamboo `${bamboo.planRepository.name}` ?